src/ArrayHelper.php

```diff
@@ -1043,24 +1047,56 @@ public static function pathExists(
      */
     public static function htmlEncode(iterable $data, bool $valuesOnly = true, ?string $encoding = null): array
     {
-        $d = [];
+        if (!is_array($data)) {
+            $data = iterator_to_array($data);
+        }
+
+        $encoding ??= ini_get('default_charset') ?: 'UTF-8';
+        $flags = ENT_QUOTES | ENT_SUBSTITUTE;
+
+        return $valuesOnly
+            ? self::htmlEncodeValues($data, $encoding, $flags)
+            : self::htmlEncodeKeysAndValues($data, $encoding, $flags);
+    }
+
+    /**
+     * @param array $data
+     * @param string $encoding
+     * @param int $flags
+     * @return array
+     */
+    private static function htmlEncodeValues(array $data, string $encoding, int $flags): array
```

Comment:
Change order of parameters to `$data, $flags, $encoding` for consistency with `htmlspecialchars()`.